nilearn/datasets.py

```diff
@@ -83,27 +91,38 @@ def _chunk_report_(bytes_so_far, total_size, t0):
     bytes_so_far: int
         Number of downloaded bytes
 
-    total_size: int, optional
-        Total size of the file. None is valid
+    total_size: int
+        Total size of the file (may be 0/None, depending on download method).
+
+    t0: int
+        The time in seconds (as returned by _get_time()) at which the
+        download was resumed / started.
 
-    t0: int, optional
-        The time in seconds (as returned by time.time()) at which the
-        download was started.
+    initial_size: int
+        If resuming, indicate the initial size of the file.
+        If not resuming, set to zero.
     """
-    if total_size:
-        percent = float(bytes_so_far) / total_size
-        percent = round(percent * 100, 2)
-        dt = time.time() - t0
-        # We use a max to avoid a division by zero
-        remaining = (100. - percent) / max(0.01, percent) * dt
-        # Trailing whitespace is too erase extra char when message length
+
+    if not total_size:
+        sys.stderr.write("Downloaded %d of ? bytes\r" % (bytes_so_far))
+
+    else:
+        # Estimate remaining download time
+        total_percent = float(bytes_so_far) / total_size
+
+        current_download_size = bytes_so_far - initial_size
+        bytes_remaining = total_size - bytes_so_far
+        dt = _get_time() - t0
+        download_rate = current_download_size / float(dt)
+        # Minimum rate of 0.01 bytes/s, to avoid dividing by zero.
+        time_remaining = bytes_remaining / max(0.01, download_rate)
```

Comment:
I think this better wait for a different system, where:
* callbacks are not based on getting packets, but rather on a timer (otherwise, if you get hung on a packet, you get no update)
* we keep track of only the most recent packets; otherwise, if you download 300MB at a fast speed, it'll be a long time before we report being "stalled" (since the current scheme is cumulative over the entire download)